File: app/src/main/java/com/example/noteapp/data/NoteSource.java
package com.example.noteapp.data;

public interface NoteSource {
    NoteSource init(NoteSourceResponse noteSourceResponse);
    Note getNote(int position);
    int size();
    void deleteNote(Note note);
    void updateNote(int position, Note note);
    void addNote(Note note);

}

File: app/src/main/java/com/example/noteapp/data/NoteSource.kt
package com.example.noteapp.data

interface NoteSource {
    fun init(noteSourceResponse: NoteSourceResponse): NoteSource
    fun getNote(position: Int): Note
    fun size(): Int
    fun deleteNote(note: Note)
    fun updateNote(position: Int, note: Note)
    fun addNote(note: Note)
}